cmd/internal/labeler/labeler.go

```diff
@@ -455,8 +471,97 @@ func createNumaNodeMappingLabel(mapping map[int][]string) string {
 	return strings.Join(parts, "_")
 }
 
-func (l *labeler) printLabels() {
+func (l *labeler) printLabelsToFile(labelFile string) error {
+	f, err := os.OpenFile(labelFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		klog.Warning("Failed to open file:", err)
+
+		return err
+	}
+
+	for key, val := range l.labels {
+		if _, err := f.WriteString(key + "=" + val + "\n"); err != nil {
+			klog.Warning("Failed to write label to file:", err)
+
+			return err
+		}
+	}
+
+	return nil
+}
+
+func CreateAndPrintLabels(sysfsDRMDir, debugfsDRIDir string) {
+	l := newLabeler(sysfsDRMDir, debugfsDRIDir)
+
+	if err := l.createLabels(); err != nil {
+		klog.Warningf("failed to create labels: %+v", err)
+
+		return
+	}
+
 	for key, val := range l.labels {
 		fmt.Println(key + "=" + val)
 	}
 }
+
+// Gathers node's GPU labels on channel trigger or timeout, and write them to a file.
+// The created label file is deleted on exit (process dying).
+func Run(pathPrefix string, updateInterval time.Duration, scanResources chan bool) {
+	l := newLabeler(pathPrefix+sysfsDRMDirectory, pathPrefix+debugfsDRIDirectory)
+
+	interruptChan := make(chan os.Signal, 1)
+	signal.Notify(interruptChan, os.Interrupt)
+	signal.Notify(interruptChan, syscall.SIGTERM)
+
```

Comment:
You could catch also `SIGHUP` & `SIGQUIT` (`os.Interrupt` aliases `SINGINT`), and all of them can be given for the same `Notify()` call, you do not need separate calls.

Reply:
Sure, though we really only need the SIGTERM as that's the one containers are put down with.

Reply:
SIGINT is useful for manual testing directly from console, and SIGQUIT is similar signal.  SIGHUP could be reserved for "re-read config" signaling, like is done with traditional daemons, though.